### PR TITLE
Introduce controller quirk handler

### PR DESCRIPTION
**Pull request by kakra**
Fixes: https://github.com/atar-axis/xpadneo/issues/192

### CODE REVIEW

**Comment by kakra** [comment]
@adam-becker Would you like to review for any obvious style and logic errors I may have missed?

File: hid-xpadneo/src/hid-xpadneo.c [hunk 1]
```diff
@@ -321,31 +381,79 @@ static int xpadneo_ff_play(struct input_dev *dev, void *data, struct ff_effect *
...
-
-	ff_pck.ff.enable = FF_RUMBLE_WEAK;
-	hid_hw_output_report(hdev, (u8 *)&ff_pck, sizeof(ff_pck));
-	mdelay(330);
+	if ((xdata->quirks & XPADNEO_QUIRK_NO_MOTOR_MASK) == 0) {
+		ff_pck.ff.magnitude_weak = 40;
+		ff_pck.ff.magnitude_strong = 20;
+		ff_pck.ff.magnitude_right = 10;
+		ff_pck.ff.magnitude_left = 10;
+	}
```
Comment (adam-becker) [inline hid-xpadneo/src/hid-xpadneo.c:395]: Could use a new line here

File: hid-xpadneo/src/hid-xpadneo.c [hunk 2]
```diff
@@ -321,31 +381,79 @@ static int xpadneo_ff_play(struct input_dev *dev, void *data, struct ff_effect *
...
+	if (xdata->quirks & XPADNEO_QUIRK_NO_MOTOR_MASK)
+		ff_pck.ff.magnitude_weak = 0;
+	else
+		ff_pck.ff.enable = 0;
+	if (xdata->quirks & XPADNEO_QUIRK_NO_PULSE) {
+		u8 save = ff_pck.ff.magnitude_weak;
+		hid_hw_output_report(hdev, (u8 *)&ff_pck, sizeof(ff_pck));
+		ff_pck.ff.magnitude_weak = save;
+	}
+	mdelay(30);
```
Comment (adam-becker) [inline hid-xpadneo/src/hid-xpadneo.c:417]: A lot of this is very compacted together, I generally separate control flow statements with whitespace wherever possible and group other statements together if they logically belong together.

File: hid-xpadneo/src/hid-xpadneo.c [hunk 3]
```diff
@@ -321,31 +381,79 @@ static int xpadneo_ff_play(struct input_dev *dev, void *data, struct ff_effect *
...
+		ff_pck.ff.magnitude_strong = save;
+	}
+	mdelay(30);
+
+	if ((xdata->quirks & XPADNEO_QUIRK_NO_TRIGGER_RUMBLE) == 0) {
+		if (xdata->quirks & XPADNEO_QUIRK_NO_MOTOR_MASK) {
+			ff_pck.ff.magnitude_left = 10;
+			ff_pck.ff.magnitude_right = 10;
+		} else
+			ff_pck.ff.enable = FF_RUMBLE_TRIGGERS;
```
Comment (adam-becker) [inline hid-xpadneo/src/hid-xpadneo.c:441]: The style guide says if either an `if` or an `else` have braces, then both have to have them.

File: hid-xpadneo/src/hid-xpadneo.c (hunk 2 above)
Comment (kakra) [reply hid-xpadneo/src/hid-xpadneo.c:417]: I'm not quite happy with that, too. But as far as the rumble testing goes, these belong together, so I didn't split it by newline. The underlying problem is that it's quite repetitive and there's no easy solution in pure C. In other programming language I would solve that by using lambdas.

I think, we'll leave it as-is until some future idea or future change allows dry'ing it. Maybe we could defer that to the `xpadneo_ff_worker` but I'd need to re-factor that to get rid of the worker struct there.

**Comment by kakra** [comment]
Merged as successfully tested.

**Pull request merged**